st3/sublime_lib/fancy_settings.py

```diff
@@ -0,0 +1,87 @@
+from uuid import uuid4
+
+def isiterable(obj):
+    try:
+        iter(obj)
+        return True
+    except TypeError:
+        return False
+
+NOT_GIVEN = {}
+
+class FancySettings():
+    def __init__(self, settings, defaults={}):
+        self.settings = settings
+        self.defaults = defaults
+
+    def __getitem__(self, key):
+        if key in self or key in self.defaults:
+            return self.get(key)
+        else:
+            raise KeyError(key)
+
+    def __setitem__(self, key, value):
+        self.settings.set(key, value)
+
+    def __delitem__(self, key):
+        if self.has(name):
+            self.settings.erase(name)
+        else:
+            raise KeyError(name)
+
+    def __contains__(self, item):
+        return self.settings.has(item)
+
+    def get(self, key, default=None):
+        return self.settings.get(key, self.defaults.get(key, default))
+
+    def pop(self, key, default=NOT_GIVEN):
+        if key in self:
+            ret = self.get(key)
+            self.erase(key)
+            return ret
+        elif default is NOT_GIVEN:
+            raise KeyError(key)
+        else:
+            return default
+
+    def setdefault(key, default=None):
+        if key in self:
+            return self.get(key)
+        else:
+            self.set(key, default)
+            return default
+
+    def update(*pairs, **kwargs):
```

Comment:
This implementation of `update` is not correct.

It accepts either a **single** parameter that is a `dict` or a sequence of key-value pairs, or no positional parameter.
It then applies updates provided through kwargs.

Reply:
Fixed.